### PR TITLE
Use tibble::print.tbl() for printing

### PR DESCRIPTION
**Pull request by krlmlr**
now only need to override `tbl_sum()`. Also fixes tests.

Closes tidyverse/dplyr#2749.

@hadley: Could you please review (and enable reviews for this repo)?

### CODE REVIEW

**Comment by hadley** [comment]
Can you keep the extra spaces so the colons line up?

File: tests/testthat/output/iris--70.txt [hunk 1]
```diff
@@ -1,9 +1,8 @@
-Source:     table<iris-output-test> [?? x 5]
-Database:   sqlite x.y.z [:memory:]
-Grouped by: Species
-Ordered by: Sepal.Length
-
-# S3: tbl_dbi
+# A tibble  : ?? x 5
+# Source    : table<iris-output-test> [?? x 5]
```
Comment (krlmlr) [inline tests/testthat/output/iris--70.txt]: Now aligning colons by adding space before them. Let me know if you prefer adding space after the colon, like in the original output.

Comment (krlmlr) [inline tests/testthat/output/iris--70.txt]: Now aligning colons by adding space before them. Let me know if you prefer adding space after the colon, like in the original output.

File: tests/testthat/output/iris--70.txt [hunk 2]
```diff
@@ -1,9 +1,8 @@
-Source:     table<iris-output-test> [?? x 5]
-Database:   sqlite x.y.z [:memory:]
-Grouped by: Species
-Ordered by: Sepal.Length
-
-# S3: tbl_dbi
+# A tibble  : ?? x 5
```
Comment (krlmlr) [inline tests/testthat/output/iris--70.txt]: Do we need to keep this?

Comment (hadley) [inline tests/testthat/output/iris--70.txt]: I don't think so

File: tests/testthat/output/iris--70.txt (hunk 1 above)
Comment (hadley) [inline tests/testthat/output/iris--70.txt]: I prefer space after colon

**Comment by krlmlr** [comment]
I don't have permission to merge.

**Comment by javierluraschi** [comment]
@krlmlr See https://github.com/tidyverse/tibble/issues/285, `dplyr` used to trigger `tbl_desc()` for computing the headers of a `tibble`, but now we rely on `tibble` itself, which is great; however, `tibble` calls `nrow` and `dim` which is subclassed in some backends to trigger a full table scan.

**Pull request merged**